hadoop-ozone/dist/src/main/smoketest/omha/om-leader-transfer.robot

```diff
@@ -14,60 +14,137 @@
-    ${split1}               ${split2}               Split String            ${omLine}           :
-    ${leaderOM} =           Strip String            ${split1}
-                            LOG                     Leader OM: ${leaderOM}
-    [return]                ${leaderOM}
+    ${result} =        Get OM Nodes
+    [Return]           ${result}[0]
 
 Get One OM Follower Node
-    ${result} =             Execute                 ozone admin om roles --service-id=omservice
-                            LOG                     ${result}
-                            Should Contain          ${result}               LEADER              1
-                            Should Contain          ${result}               FOLLOWER            2
-    ${omLines} =            Get Lines Containing String                     ${result}           FOLLOWER
-    ${omLine} =             Get Line                ${omLines}              0
-    ${split1}               ${split2}               Split String            ${omLine}           :
-    ${followerOM} =         Strip String            ${split1}
-                            LOG                     Follower OM: ${followerOM}
-    [return]                ${followerOM}
-
-
-** Test Cases ***
-Transfer Leadership for OM
+    ${result} =        Get OM Nodes
+    [Return]           ${result}[1]
+
+Get OM Leader and One Follower Node
+    ${result} =        Get OM Nodes
+    [Return]           ${result}[0]      ${result}[1]
+
+Assert OM leader Role Transitions
+    [arguments]             ${leaderOM}     ${followerOM}     ${isEqualCheck}
+    ${newLeaderOM} =        Get OM Leader Node
+                            Should not be Equal     ${leaderOM}             ${newLeaderOM}
+    Run Keyword If          '${isEqualCheck}' == 'true'         Should be Equal    ${followerOM}    ${newLeaderOM}
+
+
+*** Test Cases ***
+Transfer Leadership for OM with Valid ServiceID Specified
     # Find Leader OM and one Follower OM
-    ${leaderOM} =           Get OM Leader Node
-                            LOG                     Leader OM: ${leaderOM}
-    ${followerOM} =         Get One OM Follower Node
-                            LOG                     Follower OM: ${followerOM}
+    ${leaderOM}  ${followerOM} =  Get OM Leader and One Follower Node
+
     # Transfer leadership to the Follower OM
     ${result} =             Execute                 ozone admin om transfer --service-id=omservice -n ${followerOM}
-                            LOG                     ${result}
                             Should Contain          ${result}               Transfer leadership successfully
 
-    ${newLeaderOM} =        Get OM Leader Node
-                            Should be Equal         ${followerOM}           ${newLeaderOM}
+    Assert OM Leader Role Transitions    ${leaderOM}   ${followerOM}   true
+
+Transfer Leadership for OM with Multiple ServiceIDs, Valid ServiceID Specified
```

Comment:
_Future_ improvement idea: `Transfer Leadership for OM` is the same for all test cases.  Since they are run in context of this script, I think we can omit it.  This lets us avoid truncation of case-specific details (config, arguments):

```
Om-Leader-Transfer :: Smoketest for OM leader transfer                        
==============================================================================
Transfer Leadership for OM with Valid ServiceID Specified             | PASS |
------------------------------------------------------------------------------
Transfer Leadership for OM with Multiple ServiceIDs, Valid Service... | PASS |
------------------------------------------------------------------------------
Transfer Leadership for OM with Multiple ServiceIDs, Unconfigured ... | PASS |
------------------------------------------------------------------------------
Transfer Leadership for OM with Multiple ServiceIDs, Invalid Servi... | PASS |
------------------------------------------------------------------------------
Transfer Leadership for OM without ServiceID specified                | PASS |
------------------------------------------------------------------------------
Transfer Leadership for OM with Multiple ServiceIDs, No ServiceID ... | PASS |
------------------------------------------------------------------------------
Transfer Leadership for OM randomly with Valid ServiceID Specified    | PASS |
------------------------------------------------------------------------------
Transfer Leadership for OM randomly with Multiple ServiceIDs, Vali... | PASS |
------------------------------------------------------------------------------
Transfer Leadership for OM randomly with Multiple ServiceIDs, Unco... | PASS |
------------------------------------------------------------------------------
Transfer Leadership for OM randomly with Multiple ServiceIDs, Inva... | PASS |
------------------------------------------------------------------------------
Transfer Leadership for OM randomly without ServiceID specified       | PASS |
------------------------------------------------------------------------------
Transfer Leadership for OM randomly with Multiple ServiceIDs, No S... | PASS |
------------------------------------------------------------------------------
Om-Leader-Transfer :: Smoketest for OM leader transfer                | PASS |
12 tests, 12 passed, 0 failed
```

Reply:
Sure, sounds reasonable.